tsl/test/expected/recompress_chunk_segmentwise.out

```diff
@@ -0,0 +1,419 @@
+----------------------------------------------------------
+ Append
+   ->  Custom Scan (DecompressChunk) on _hyper_9_10_chunk
+         ->  Seq Scan on compress_hyper_10_11_chunk
+   ->  Seq Scan on _hyper_9_10_chunk
+(4 rows)
+
+EXECUTE p1;
+             time             | a | b | c 
+------------------------------+---+---+---
+ Sun Jan 01 00:00:00 2023 PST | 2 |   | 2
+ Sun Jan 01 00:00:00 2023 PST | 2 |   | 2
+ Sun Jan 01 00:00:00 2023 PST | 2 | 3 | 2
+(3 rows)
+
+-- check plan again after recompression
+CALL recompress_chunk(:'chunk_to_compress_prep');
+EXPLAIN (COSTS OFF) EXECUTE p1;
+                     QUERY PLAN                     
+----------------------------------------------------
+ Custom Scan (DecompressChunk) on _hyper_9_10_chunk
+   ->  Seq Scan on compress_hyper_10_11_chunk
+(2 rows)
+
+EXECUTE p1;
+             time             | a | b | c 
+------------------------------+---+---+---
+ Sun Jan 01 00:00:00 2023 PST | 2 |   | 2
+ Sun Jan 01 00:00:00 2023 PST | 2 |   | 2
+ Sun Jan 01 00:00:00 2023 PST | 2 | 3 | 2
+(3 rows)
+
+-- verify segmentwise recompression when index exists, decompress + compress otherwise
+CREATE TABLE mytab (time timestamptz, a int, b int, c int);
+SELECT create_hypertable('mytab', 'time');
+NOTICE:  adding not-null constraint to column "time"
+  create_hypertable  
+---------------------
+ (11,public,mytab,t)
+(1 row)
+
+INSERT INTO mytab VALUES ('2023-01-01'::timestamptz, 2, NULL, 2),
+('2023-01-01'::timestamptz, 2, NULL, 2);
+select show_chunks as chunk_to_compress_mytab from show_chunks('mytab') limit 1 \gset
+-- index exists, recompression should happen segment by segment so expect a  debug message
+alter table mytab set (timescaledb.compress, timescaledb.compress_segmentby = 'a, c');
+select compress_chunk(show_chunks('mytab'));
+              compress_chunk              
+------------------------------------------
+ _timescaledb_internal._hyper_11_12_chunk
+(1 row)
+
+INSERT INTO mytab VALUES ('2023-01-01'::timestamptz, 2, 3, 2);
+set client_min_messages = DEBUG;
+call recompress_chunk(:'chunk_to_compress_mytab');
+LOG:  statement: call recompress_chunk('_timescaledb_internal._hyper_11_12_chunk');
+DEBUG:  performing segmentwise recompression of chunk _hyper_11_12_chunk in single transaction
+DEBUG:  Index oid for recompress is 17303
+set client_min_messages = WARNING;
+LOG:  statement: set client_min_messages = WARNING;
+SELECT decompress_chunk(show_chunks('mytab'));
+             decompress_chunk             
+------------------------------------------
+ _timescaledb_internal._hyper_11_12_chunk
+(1 row)
+
+alter table mytab set (timescaledb.compress = false);
+alter table mytab set (timescaledb.compress);
+select compress_chunk(show_chunks('mytab'));
+              compress_chunk              
+------------------------------------------
+ _timescaledb_internal._hyper_11_12_chunk
+(1 row)
+
+INSERT INTO mytab VALUES ('2023-01-01'::timestamptz, 2, 3, 2);
+set client_min_messages = DEBUG;
```

Comment:
This will lead to a flaky tests as you will randomly get DEBUG messages from other postgres subsystems. Can we change the test to not use DEBUG log level?

Reply:
I changed the test to verify which method is used for recompression by checking whether the compressed chunk is the existing one or a new one after the operation completes.